main/src/ca/uwaterloo/flix/language/phase/Weeder.scala

```diff
@@ -457,18 +457,19 @@ object Weeder {
   private def visitImport(i0: ParsedAst.Import): Validation[List[WeededAst.Import], WeederError] = i0 match {
     case ParsedAst.Imports.ImportOne(sp1, name, sp2) =>
       val loc = mkSL(sp1, sp2)
-      val ident = Name.Ident(sp1, name.fqn.last, sp2)
-      List(WeededAst.Import.Import(name, ident, loc)).toSuccess
+      val alias = name.fqn.last
+      if (raw"[A-Z][A-Za-z0-9_?]*".r matches alias) {
```

Comment:
Do we actually allow `?` in our names?

Reply:
Oops! That was meant to be `!` (blush). A good example of why it would be nice to be able to reuse the code in the Parser (I don't suppose you're aware of any way I could do this?)

Reply:
I don't think the Parboiled API makes it easy to do that. The macros seem to enforce some kind of self-containment of the parser which helps performance but hurts reusability.